zdm/survey.py

```diff
@@ -103,18 +104,24 @@ def process_survey_file(self,filename):
         #### Find the number of FRBs in the file ###
         self.info=info
         self.keys=keys
-        #try:
-        #	iNFRB=keys.index('NFRB')
-        #except:
-        self.NFRB=keys.count('FRB')
+
+        # 
+        if NFRB is None:
+            self.NFRB=keys.count('FRB')
+        else:
+            self.NFRB = NFRB
         if self.NFRB==0:
             raise ValueError('No FRBs found in file '+filename) #change this?
```

Comment:
For now - leave it there. In the future, we want to re-write this such that you can have a survey with no results, just survey parameters. Although clearly such a survey could not be fit to, one could clearly create a zdm grid for it (provided a mean Galactic DM component was used)

Reply:
Roger